multiversx_sdk_core/transaction_factories/delegation_factory.py

```diff
@@ -0,0 +1,408 @@
+                     gas_price: Optional[IGasPrice] = None,
+                     gas_limit: Optional[IGasLimit] = None) -> Transaction:
+        num_nodes = len(bls_keys)
+
+        parts = ["unJailNodes"]
+        parts.extend(bls_keys)
+
+        transaction = self.create_transaction(
+            sender=sender,
+            receiver=delegation_contract,
+            data_parts=parts,
+            execution_gas_limit=self.config.delegation_ops + num_nodes * self.config.additional_gas_limit_per_node,
+            gas_limit_hint=gas_limit,
+            gas_price=gas_price,
+            nonce=transaction_nonce,
+            value=value
+        )
+
+        return transaction
+
+    def change_service_fee(self,
+                           sender: IAddress,
+                           delegation_contract: IAddress,
+                           service_fee: int,
+                           value: ITransactionValue,
+                           transaction_nonce: Optional[INonce] = None,
+                           gas_price: Optional[IGasPrice] = None,
+                           gas_limit: Optional[IGasLimit] = None) -> Transaction:
+        parts = [
+            "changeServiceFee",
+            arg_to_string(service_fee)
+        ]
+
+        transaction = self.create_transaction(
+            sender=sender,
+            receiver=delegation_contract,
+            data_parts=parts,
+            execution_gas_limit=self.config.delegation_ops + self.config.additional_gas_for_operations,
+            gas_limit_hint=gas_limit,
+            gas_price=gas_price,
+            nonce=transaction_nonce,
+            value=value
+        )
+
+        return transaction
+
+    def modify_delegation_cap(self,
+                              sender: IAddress,
+                              delegation_contract: IAddress,
+                              delegation_cap: int,
+                              value: ITransactionValue,
+                              transaction_nonce: Optional[INonce] = None,
+                              gas_price: Optional[IGasPrice] = None,
+                              gas_limit: Optional[IGasLimit] = None) -> Transaction:
+        parts = [
+            "modifyTotalDelegationCap",
+            arg_to_string(delegation_cap)
+        ]
+
+        transaction = self.create_transaction(
+            sender=sender,
+            receiver=delegation_contract,
+            data_parts=parts,
+            execution_gas_limit=self.config.delegation_ops + self.config.additional_gas_for_operations,
+            gas_limit_hint=gas_limit,
+            gas_price=gas_price,
+            nonce=transaction_nonce,
+            value=value
+        )
+
+        return transaction
+
+    def automatic_activation(self,
+                             sender: IAddress,
+                             delegation_contract: IAddress,
+                             set: bool,
+                             unset: bool,
```

Comment:
`set` and `unset` should be unified into a single parameter, something like `status_value` that can be a bool and directly appended to arg parts. Otherwise, in the current form, you can get a malformed transaction if user sets both these parameters as true (this will lead to `parts=[setAutomaticActivation, true, false]`). Plus is confusing cause you get the impression that you have to provide both.

Reply:
As discussed, I've split the method into 2 separate methods called `set_automatic_activation` and `unset_automatic_activation`